src/backend/metal/src/device.rs

```diff
@@ -752,11 +762,13 @@ impl hal::Device<Backend> for Device {
 
         let mut num_samplers = 0;
         let mut num_textures = 0;
+        let mut num_uniforms = 0;
```

Comment:
I don't know what's the point of these, @kvark

Reply:
To count resources for allocation in an argument buffer, see code below